src/app/core/models/label.model.ts

```diff
@@ -1,4 +1,74 @@
 export interface Label {
-    labelValue: string;
-    labelColor: string;
-  }
+  labelValue: string;
+  labelColor: string;
+  labelCategory: LABEL_CATEGORY;
+}
+
+export enum LABEL_CATEGORY {
+  severity = 'severity',
+  type = 'type',
+  response = 'response',
+  status = 'status'
+}
+
+export enum CATEGORY_STATUS_VALUES {
+  Incomplete = 'Incomplete',
+  Done = 'Done',
+}
+
+export enum CATEGORY_SEVERITY_VALUES {
+  Low = 'Low',
+  Medium = 'Medium',
+  High = 'High',
+}
+
+export enum CATEGORY_TYPE_VALUES {
+  DocumentationBug = 'DocumentationBug',
+  FunctionalityBug = 'FunctionalityBug',
+}
+
+export enum CATEGORY_RESPONSE_VALUES {
+  Accepted = 'Accepted',
+  Rejected = 'Rejected',
+  IssueUnclear = 'IssueUnclear',
+  CannotReproduce = 'CannotReproduce',
+}
+
+export const LABEL_VALUES = {
+  severity: CATEGORY_SEVERITY_VALUES,
+  type: CATEGORY_TYPE_VALUES,
+  response: CATEGORY_RESPONSE_VALUES,
+  status: CATEGORY_STATUS_VALUES
+};
+
+export const DEFAULT_LABEL_COLOR = 'ededed';
+
+enum STATUS_COLORS {
+  Incomplete = '1ac6ff',
+  Done = 'b3ecff'
+}
+
+enum SEVERITY_COLORS {
+  Low = 'ffb3b3',
+  Medium = 'ff6666',
+  High = 'b30000'
+}
+
+enum TYPE_COLORS {
+  DocumentationBug = 'ccb3ff',
+  FunctionalityBug = '661aff'
+}
+
+enum RESPONSE_COLORS {
+  Accepted = '80ffcc',
+  Rejected = 'ff80b3',
+  IssueUnclear = 'ffcc80',
+  CannotReproduce = 'bfbfbf'
+}
+
+export const LABEL_COLORS = {
+  status: STATUS_COLORS,
+  severity: SEVERITY_COLORS,
+  type: TYPE_COLORS,
+  response: RESPONSE_COLORS
+};
```

Comment:
Looks like there are many enums. Is it possible to consolidate using a more sophisticated structure. Example:
```js
private severityLabels= {
  Low: Label('severity', 'Low', 'ffb3b3'),
  Medium: Label('severity', 'Medium', 'abcabc'),
  ...
};
```

Reply:
I think this is possible.

@KevinCJH @JunWei96 @RonakLakhotia Would it be okay to keep the Required Labels as Label Objects or store their properties as enums?

Reply:
I think it would be cleaner to keep them as Label Objects instead of enums. Also, remember to keep track of the rank of the labels so that the `dataTable` can sort the information according to the labels correctly.